library/src/c_api/mod.rs

```diff
@@ -212,6 +237,16 @@ pub unsafe extern "C" fn shorebird_free_string(c_string: *mut c_char) {
     }
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn shorebird_free_update_result(result: *mut UpdateResult) {
+    if result.is_null() {
+        return;
+    }
+    unsafe {
+        drop(Box::from_raw(result));
```

Comment:
This isn't sufficient.  We also need to free the string.

Reply:
yeah was wondering about that kk thanks will fix

Reply:
There are other ways we could do this if we *didn't* need to release that string.  We could change the signature to have the *Dart* create the Status object for us (and pass it in as a * or ** to update).  But then we'd still need Dart to know to call free_string.  That might be less code if you'd prefer that method.

Reply:
I updated to free the string conditionally if it's not null